File: ios/Model/EditorPreset.swift
/// An enum containing all prebuild editor configurations.
@objc public enum EditorPreset: Int {
  /// The apparel editor.
  case apparel

  /// The postcard editor.
  case postcard

  /// The photo editor.
  case photo

  /// The video editor.
  case video

  /// The design editor.
  case design

  // Converts enum case to the corresponding string
  public var stringValue: String {
    switch self {
    case .apparel: "apparel"
    case .postcard: "postcard"
    case .photo: "photo"
    case .video: "video"
    case .design: "design"
    }
  }

  // Initialize from string
  public static func fromString(_ rawValue: String) -> EditorPreset {
    switch rawValue {
    case "apparel": .apparel
    case "postcard": .postcard
    case "photo": .photo
    case "video": .video
    case "design": .design
    default: .design
    }
  }
}

@objc @objcMembers public class EditorPresetParser: NSObject {
  // Method to convert from NSString to EditorPreset enum
  public static func fromString(_ rawValue: NSString) -> EditorPreset {
    EditorPreset.fromString(rawValue as String)
  }

  // Method to convert from EditorPreset enum to NSString
  public static func toString(_ preset: EditorPreset) -> NSString {
    preset.stringValue as NSString
  }
}

File: android/src/main/kotlin/ly/img/editor/reactnative/module/model/EditorPreset.kt
package ly.img.editor.reactnative.module.model

import android.os.Parcel
import android.os.Parcelable

/** An enum containing all prebuilt editor configurations. */
enum class EditorPreset(
    val value: String,
) : Parcelable {
    /** The apparel editor. */
    APPAREL("apparel"),

    /** The postcard editor. */
    POSTCARD("postcard"),

    /** The photo editor. */
    PHOTO("photo"),

    /** The design editor. */
    DESIGN("design"),

    /** The video editor. */
    VIDEO("video"),
    ;

    override fun toString(): String = value

    override fun writeToParcel(
        parcel: Parcel,
        flags: Int,
    ) {
        parcel.writeString(value)
    }

    override fun describeContents(): Int = 0

    companion object CREATOR : Parcelable.Creator<EditorPreset> {
        override fun createFromParcel(parcel: Parcel): EditorPreset {
            val value = parcel.readString()
            return entries.find { it.value == value } ?: throw IllegalArgumentException("Invalid value for EditorPreset")
        }

        override fun newArray(size: Int): Array<EditorPreset?> = arrayOfNulls(size)

        fun fromValue(value: String): EditorPreset? = entries.find { it.value == value }
    }
}
